games/space-invaders/Space Invaders [David Winter].json

```diff
@@ -0,0 +1,108 @@
+{
+  "title": "Space Invaders",
+  "authors": [
+    {
+      "id": "david-winter",
+      "name": "David Winter"
+    }
+  ],
+  "release": "1978",
+  "category": "GAME",
+  "description": "The well known game. Destroy the invaders with your ship.",
+  "images": [
+    "./Space Invaders [David Winter] - main screen.png",
+    "./Space Invaders [David Winter] - game play.png"
+  ],
+  "interpreter": {
+    "id": "chip8-david-winter",
+    "name": "CHIP8 by David Winter"
+  },
+  "binaries": {
+    "names": [
+      { "value": "Space Invaders [David Winter] - 0.9.1" },
+      { "value": "Space Invaders [David Winter]" },
+
+      { "value": "Space Invaders [David Winter] - 0.9.0" },
+      { "value": "Space Invaders [David Winter] (alt)" },
+      { "value": "INVADERS" }
+    ],
+    "extensions": [
+      "c8",
+      "ch8",
+      ""
+    ],
+    "hashes": [
+      {
+        "type": "SHA256",
+        "value": "00bb7001de52b562d98357d33375406c6d0308466d7d95c2580537d2451a75cd"
+      },
+      {
+        "type": "SHA256",
+        "value": "2d0e1fa53216b297e74041d4fb766f42327a42893e83bb4ec931a9dff5c2dd10"
+      }
+    ]
+  },
+  "sources": {
+    "names": [
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.0"
+      },
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.1"
+      }
+    ],
+    "extensions": [
+      "8o",
+      "asm",
+      "txt",
+      ""
+    ]
+  },
+  "platform": {
+    "name": "chip8",
+    "startAddress": "0x200",
+    "resolution": {
+      "width": 64,
+      "height": 32
+    }
+  },
+  "keys": [
+    {
+      "label": "Start",
+      "action": "START",
+      "stroke": "ENTER",
+      "value": "0x5"
+    },
+    {
+      "label": "Fly left",
+      "action": "LEFT",
+      "stroke": "LEFT",
+      "value": "0x4"
+    },
+    {
+      "label": "Fly right",
+      "action": "RIGHT",
+      "stroke": "RIGHT",
+      "value": "0x6"
+    },
+    {
+      "label": "Shoot",
+      "action": "X",
+      "stroke": "SPACE",
+      "value": "0x5"
+    }
+  ],
+  "options": {
+    "opsPerSec": 1500,
```

Comment:
This changed from `tickrate` which in README is described as 
> how many cycles to run per frame

The problem with this is chip8 doesn't have interrupts which would draw a boundary between frames. Instead it draws continuously. Does it mean that every draw operation is a frame? I don't think so. I implemented a kind of frame detection but 99% of emulators don't have that.

I was considering `frequency` but chip8 is VM running on a host CPU which has some frequency but it's not really related. There are efforts to map chip8 instruction cycles to COSMAC VIP cycles but it's only one of interpreters out there.

I think "operations per second" is a pretty good name since most of the interpreters implement frequency / tickrate like that (ignoring frames or how many COSMAC VIP cycles given instruction takes).

Reply:
Sounds very logical to me. Not all interpreters run at 60 frames per second either – DREAM 6800 and ETI-660 were Australian computers and ran at 50 Hz.